rfcs/build-system/02_standard_package_output_folders.md

```diff
@@ -0,0 +1,45 @@
+# RFC: Standard package output folders
+
+---
+
+_List contributors to the proposal: @dzearing_
+
+## Summary
+
+There are multiple javascript formats we may include in our npm packages: `commonjs`, `esm`, and `amd`, in addition to static assets like pre-made bundles and images.
+
+We should adopt a predictable folder structure within the published npm package for these formats. That way consumers of our packages can know exactly where things are and which folders have what things in as few steps as possible.
+
+## Problem statement
+
+Recently a change to a library to use the `build:commonjs-only` build step changed the `commonjs` output folder from `lib-commonjs` to `lib`. This in turn caused a partner to hit friction because on syncing the change, their `lib` folder still had `esm`. In debugging why they hit this, developers unfamiliar with the change didn't know why the package.json `main` entry was changed to `lib` and had to hunt down if the change was intentional or not.
```

Comment:
> This in turn caused a partner to hit friction because on syncing the change, their `lib` folder still had `esm`.

Doesn't invalidate the main point of the RFC, but I don't think a point-in-time issue like this should be cited as the motivation to standardize. The way to address that specific issue should be to ensure that our dev docs are clear on what you need to do after updating from master: currently, run `yarn` and `yarn build --to your-project` (and yes I know that the ideal/long-term solution is to update our infra so the build step isn't necessary).

Reply:
I've reworded to get to the problem faster without reflecting on the background info.